### PR TITLE
Update topolvm chart to v4.0.2 and remove custom lvmd resources

### PR DESCRIPTION
**Pull request by DebakelOrakel**
## Checklist

- [ ] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [ ] Link this PR to related issues.

### CODE REVIEW

File: class/defaults.yml [hunk 1]
```diff
@@ -76,14 +70,22 @@ parameters:
...
         # lvmd.resources -- Specify resources.
         resources: ${topolvm:resources:lvmd}
+        # lvmd.deviceClasses -- Specify the device-class settings.
+        deviceClasses: ${topolvm:deviceclasses}
+        # lvmd.nodeSelector -- Specify nodeSelector.
+        nodeSelector:
+          syn.tools/topolvm: ""
+
+      # storageClasses -- Whether to create storageclass(s)
+      storageClasses: []
```
Comment (megian) [inline class/defaults.yml:88]: There is no reference to ${topolvm:storageclasses}. Is this intended? Why is it an hash first and a list later?

File: class/defaults.yml [hunk 2]
```diff
@@ -6,25 +6,19 @@ parameters:
...
         repository: topolvm/topolvm-with-sidecar
-        tag: 0.10.4
+        tag: 0.10.6
 
-    deviceclasses:
-      ssd:
-        volume-group: vgssd
-        spare-gb: 10
-        default: true
+    deviceclasses: []
```
Comment (megian) [inline class/defaults.yml:19]: Looks valid to me https://github.com/topolvm/topolvm/blob/main/charts/topolvm/values.yaml#L134

File: class/defaults.yml [hunk 3]
```diff
@@ -6,25 +6,19 @@ parameters:
...
 
-    storageclasses:
-      ssd-local:
-        class: ssd
-        fstype: ext4
-        volumeexpansion: true
+    storageclasses: {}
```
Comment (megian) [inline class/defaults.yml:21]: Looks to be a list `[]` and not a `{}`to me https://github.com/topolvm/topolvm/blob/main/charts/topolvm/values.yaml#L430

File: docs/modules/ROOT/pages/how-tos/upgrade-v1-to-v2.adoc [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+[source,bash]
+----
+parameters:
+  topolvm:
+    deviceclasses:
+      - name: ssd
+        volume-group: vgssd
+        default: true
+        spare-gb: 10
+----
```
Comment (megian) [inline docs/modules/ROOT/pages/how-tos/upgrade-v1-to-v2.adoc:32]: I think the storageclass part also needs a similar change. https://github.com/topolvm/topolvm/blob/main/charts/topolvm/values.yaml#L430

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 5]
```diff
@@ -68,15 +39,7 @@ parameters:
...
-default::
-+
-[source,yaml]
-----
-ssd-local:
-  class: ssd
-  fstype: ext4
-  volumeexpansion: true
-----
+default:: {}
```
Comment (megian) [inline docs/modules/ROOT/pages/references/parameters.adoc:42]:
```suggestion
default:: []
```

Comment (megian) [reply docs/modules/ROOT/pages/references/parameters.adoc:42]: Probably an example like in https://github.com/topolvm/topolvm/blob/main/charts/topolvm/values.yaml#L430 could make sense.

File: tests/defaults.yml [hunk 6]
```diff
@@ -1,3 +1,15 @@
...
+# parameters:
+#   topolvm:
+#     deviceclasses:
+#       - name: ssd
+#         volume-group: vgssd
+#         default: true
+#         spare-gb: 10
+
+#     storageclasses:
+#       ssd-local:
```
Comment (megian) [inline tests/defaults.yml:12]: I think must be adapted as well https://github.com/topolvm/topolvm/blob/main/charts/topolvm/values.yaml#L430.

File: class/defaults.yml (hunk 1 above)
Comment (DebakelOrakel) [reply class/defaults.yml:88]: Yes I create the storage class using the storageclass component.
https://github.com/projectsyn/component-topolvm/blob/master/component/main.jsonnet#L131

File: class/defaults.yml (hunk 3 above)
Comment (DebakelOrakel) [reply class/defaults.yml:21]: See my comment above.

File: docs/modules/ROOT/pages/how-tos/upgrade-v1-to-v2.adoc (hunk 4 above)
Comment (DebakelOrakel) [reply docs/modules/ROOT/pages/how-tos/upgrade-v1-to-v2.adoc:32]: See my comment above

File: class/topolvm.yml [hunk 7]
```diff
@@ -2,10 +2,10 @@ parameters:
   kapitan:
     dependencies:
       - type: helm
-        source: https://topolvm.github.io/topolvm
+        source: ${topolvm:charts:topolvm:source}
         chart_name: topolvm
-        version: ${topolvm:charts:topolvm}
-        output_path: dependencies/topolvm/helmcharts/topolvm/${topolvm:charts:topolvm}/
+        version: ${topolvm:charts:topolvm:version}
+        output_path: dependencies/topolvm/helmcharts/topolvm/${topolvm:charts:topolvm:version}
```
Comment (megian) [inline class/topolvm.yml]:
```suggestion
        output_path: dependencies/topolvm/helmcharts/topolvm/${topolvm:charts:topolvm:version}/
```

File: docs/modules/ROOT/pages/how-tos/upgrade-v1-to-v2.adoc [hunk 8]
```diff
@@ -0,0 +1,32 @@
+= Upgrade from v1 to v2
+
+With the recent update of Kapitan, Helm 3 charts can be used. Therefor the need for the custom lvmd daemonset is gone.
+Therefor a minor change in the `deviceclasses` parameter is necessary.
+
+== Update component parameters
+
+Change `deviceclasses` from `dictionary` to `array`
```
Comment (megian) [inline docs/modules/ROOT/pages/how-tos/upgrade-v1-to-v2.adoc]:
```suggestion
Change `deviceclasses` from `dictionary` to `array`:
```

**Review by megian — CHANGES_REQUESTED** [review]

**Review by megian — APPROVED** [review]

**Pull request merged**